src/joystick/hidapi/SDL_hidapi_gip.c

```diff
@@ -1197,9 +1212,13 @@ static bool GIP_SendInitSequence(GIP_Attachment *attachment)
         GIP_SendVendorMessage(attachment, GIP_CMD_DEVICE_CAPABILITIES, 0, NULL, 0);
     }
 
-    if (!attachment->joystick) {
+    if ((!attachment->attachment_index || GIP_AttachmentIsController(attachment)) && !attachment->joystick) {
         return HIDAPI_JoystickConnected(attachment->device->device, &attachment->joystick);
     }
+    if (attachment->attachment_type == GIP_TYPE_CHATPAD && !attachment->keyboard) {
+        attachment->keyboard = (SDL_KeyboardID)(uintptr_t) attachment;
```

Comment:
Are you guaranteed that this won't conflict with existing and future keyboard IDs?

Reply:
Ah, since it's a pointer, this is unlikely.

Reply:
This is the same technique I found used in a few other places. However, `SDL_KeyboardID` is 32 bits, so it's not guaranteed. I can shave off the lower bits to increase the number of unique bits, but that's probably overkill. If SDL kept an internal counter that guaranteed no reuse that would be ideal, but that's not what's there right now.